packages/ui-logic/src/weave.ts

```diff
@@ -26,6 +26,9 @@ export class Weave {
         return null;
       }
     } catch {
+      // Error is not JSON
+      if (normalizedErrorMessage) return normalizedErrorMessage;
+      // Error is empty string
```

Comment:
Do you need to return error message in `try` block too instead of `null`? Like you did in `catch` block?

Reply:
I'm not sure about this whole implementation. IOV Core [was updated](https://github.com/iov-one/ponferrada/issues/787#issuecomment-536870859) to show an error when creating a Distribute Funds proposal with no recipients file.

The issue is, for some reason, that error message is not JSON, so in the frontend shows as "An unknown error ocurred" because of [this line](https://github.com/iov-one/ponferrada/blob/143f7018ee36969c00f1ee91d4b4880e8c23a6d5/packages/sil-governance/src/routes/create-proposal/components/ProposalForm.tsx#L211).

If `JSON.parse` fails it returns `null`, which makes the unknown error message appear. What I did was add support for non-JSON errors in the `ui-logic -> weave -> tryParse` function so that the right message is shown. This was achieved by returning `normalizedErrorMessage` if not empty string and if failed to parse as JSON.